### PR TITLE
Bigquery e2e testing

### PR DESCRIPTION
**Pull request by beczkowb**

### CODE REVIEW

File: e2e/test_dataset_manager.py [hunk 1]
```diff
@@ -13,12 +20,18 @@ def df_to_collections(df):
 
 
 class DatasetManagerTestCase(TestCase):
-    TEST_PARTITION = '2019-01-01'
-    TEST_PARTITION_PLUS_ONE = '2019-01-02'
+    TEST_PARTITION_DT = (datetime.utcnow() - timedelta(days=-1)).replace(hour=0, minute=0, second=0, microsecond=0, tzinfo=UTC)
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply e2e/test_dataset_manager.py:23]: I have tried to insert a record to the '2019-01-01' partition and got an error (partition too old), that's why I have changed the date.

File: bigflow/bigquery/interface.py [hunk 2]
```diff
@@ -34,9 +37,28 @@ def collect(self, sql: str) -> BigQueryOperation:
...
     @abstractmethod
     def create_table(self, create_query: str) -> BigQueryOperation:
         pass
+
+    @abstractmethod
+    def create_table_from_schema(
+            self,
+            table_name: str,
+            schema: typing.Union[dict, Path],
+            table=None):
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply bigflow/bigquery/interface.py:52]: I can't, the `google.cloud.bigquery.Table` import is inside the method (#149).

File: bigflow/bigquery/dataset_manager.py [hunk 3]
```diff
@@ -109,14 +113,29 @@ def template_variables(self, custom_run_datetime=None):
         result['dt'] = custom_run_datetime or self.run_datetime
         return result
 
+    def create_table_from_schema(
+            self,
+            table_name: str,
+            schema: typing.Optional[typing.Union[typing.List[dict], Path]] = None,
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply bigflow/bigquery/dataset_manager.py]: done

File: bigflow/bigquery/dataset_manager.py [hunk 4]
```diff
@@ -109,14 +113,29 @@ def template_variables(self, custom_run_datetime=None):
...
+        table_id = self.create_table_id(table_name)
+        return self.dataset_manager.insert(table_id, records)
+
 
 class PartitionedDatasetManager(object):
     """
     Interface available for user. Manages partitioning.
     Delegate rest of the tasks to TemplatedDatasetManager and DatasetManager.
     """
     def __init__(self, templated_dataset_manager, partition):
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply bigflow/bigquery/dataset_manager.py]: done

File: bigflow/bigquery/dataset_manager.py [hunk 5]
```diff
@@ -292,6 +331,44 @@ def table_exists(self, table_name):
...
+
+        self.logger.info(f'CREATING TABLE FROM SCHEMA: {table.schema}')
+
+        self.bigquery_client.create_table(table)
+
+    def insert(
+            self,
+            table_id: str,
+            records: typing.Union[typing.List[dict], Path]):
+        self.logger.info(f'INSERTING RECORDS TO TABLE: {table_id}')
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply bigflow/bigquery/dataset_manager.py]: done

File: bigflow/bigquery/dataset_manager.py [hunk 6]
```diff
@@ -292,6 +331,44 @@ def table_exists(self, table_name):
...
+            table=None):
+        from google.cloud.bigquery import Table, TimePartitioning
+
+        if schema and table:
+            raise ValueError("You can't provide both schema and table, because the table you provide"
+                             "should already contain the schema.")
+        if not schema and not table:
+            raise ValueError("You must provide either schema or table.")
+
+        if isinstance(schema, Path):
```
[the remark this answers is not in the document]
Comment (beczkowb) [reply bigflow/bigquery/dataset_manager.py:347]: done

**Review by anjensan — APPROVED** [review]

**Pull request merged**